### ISSUE TITLE
btc_transfer_context() has a possible defect

### ISSUE THREAD

**Issue by dadwin** [espressif/esp-idf#531]
That's from file esp-idf/components/bt/bluedroid/btc/core/btc_task.c.
The function contains that piece of code:
```
        lmsg.arg = (void *)GKI_getbuf(arg_len);
        memset(lmsg.arg, 0x00, arg_len);    //important, avoid arg which have no length
        if (lmsg.arg == NULL) {
            return BT_STATUS_NOMEM;
        }
```

GKI_getbuf() allocates memory. It makes sense to check it right after the function returns, otherwise memset() has a chance to write on NULL pointer which leads to segmentation fault.
Or correct me if I'm wrong :)

**Comment by projectgus** [comment]
This is fixed in master branch as of a4f3312d9, I think the fix was in review to be merged when this bug report was opened. Sorry for the crossover and the slow reply.

**Comment by dadwin** [comment]
Thank you!
